File: trunk/proj/CoreLibrary/src/com/jasonzqshen/familyaccounting/core/masterdata/MasterDataIdentity_GLAccount.java
package com.jasonzqshen.familyaccounting.core.masterdata;

import com.jasonzqshen.familyaccounting.core.exception.IdentityInvalidChar;
import com.jasonzqshen.familyaccounting.core.exception.IdentityNoData;
import com.jasonzqshen.familyaccounting.core.exception.IdentityTooLong;

/**
 * The identity only within numbers
 * 
 * @author I072485
 * 
 */
public class MasterDataIdentity_GLAccount extends MasterDataIdentity {
	public MasterDataIdentity_GLAccount(String id) throws IdentityTooLong,
			IdentityNoData, IdentityInvalidChar {
		super(id);
	}

	public MasterDataIdentity_GLAccount(char[] id) throws IdentityTooLong,
			IdentityNoData, IdentityInvalidChar {
		super(id);
	}

	/**
	 * only number
	 * 
	 * @param ch
	 * @return
	 */
	@Override
	protected boolean isValidChar(char ch) {
		if ('0' <= ch && ch <= '9') {
			return true;
		}
		return false;
	}

}

File: proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/MasterData/MasterDataIdentity_GLAccount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinStore_FamilyAccountingCore.MasterData
{
    /// <summary>
    /// The identity only contains numbers
    /// </summary>
    public class MasterDataIdentity_GLAccount : MasterDataIdentity
    {
        /// <summary>
        /// Constructr
        /// </summary>
        /// <param name="id">string of id</param>
        /// <exception cref="IdentityTooLong">Length of input string is longer than the limitation
        /// </exception>
        /// <exception cref="IdentityNoData">No data contain in the input string, empty or null
        /// </exception>
        /// <exception cref="IdentityInvalidChar">Invalid character in the input string.
        /// </exception>
        /// <exception cref="NullReferenceException"></exception>
        public MasterDataIdentity_GLAccount(String id)
            : base(id)
        {
        }
        /// <summary>
        /// Constructr
        /// </summary>
        /// <param name="id">char array of id</param>
        /// <exception cref="IdentityTooLong">Length of input string is longer than the limitation
        /// </exception>
        /// <exception cref="IdentityNoData">No data contain in the input string, empty or null
        /// </exception>
        /// <exception cref="IdentityInvalidChar">Invalid character in the input string.
        /// </exception>
        public MasterDataIdentity_GLAccount(char[] id)
            : base(id)
        {
        }

        /// <summary>
        /// Only number is valid.
        /// </summary>
        /// <param name="ch"></param>
        /// <returns></returns>
        protected override bool isValidChar(char ch)
        {
            if ('0' <= ch && ch <= '9')
            {
                return true;
            }
            return false;
        }
    }
}
